### PR TITLE
Kernel: Move the boot code into a separate binary

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**Kernel: Ignore KUBSAN symbols in the kernel symbol map**

**readelf: Add names for x86_64 relocations and machine type**

**Kernel: Fix {read,write}_gs_ptr() for 64-bit addresses**

**Kernel: Make SCHEDULER_DEBUG work on x86_64**

**Kernel: Remove some more hardcoded addresses**

**Kernel: Make VirtualAddress::page_base() work with 64-bit addresses**

**Kernel: Support 64-bit addresses for kernel thread stacks**

**Kernel: Add support for kernel addresses other than 3-4GB**

**Kernel: Split debug symbols into a separate file**

This speeds up the boot process considerably when specifying the kernel image via -initrd.

**pmap: Add support for 64-bit addresses**

**SystemMonitor: Add support for 64-bit addresses**

**Kernel: Fix 64-bit address truncation in MemoryManager::ensure_pte()**

**Kernel: Make allocate_randomized() work for 64-bit addresses**

The odds of finding a suitable address in 1000 attempts were not in our favor given the size of the 64-bit address space.

**Kernel: Introduce basic pre-kernel environment**

This implements a simple bootloader that is capable of loading ELF64 kernel images. It does this by using QEMU/GRUB to load the kernel image from disk and pass it to our bootloader as a Multiboot module.

The bootloader then parses the ELF image and sets it up appropriately. The kernel's entry point is a C++ function with architecture-native code.

**Kernel: Use a different kernel load address for x86_64**

**Kernel: Hand out 64-bit addresses to userspace**

### CODE REVIEW

File: Kernel/linker_amd64.ld [hunk 1]
```diff
@@ -0,0 +1,90 @@
+ENTRY(init)
+
+KERNEL_VIRTUAL_BASE = 0x2000000000;
```
Comment (Hendiadyoin1) [inline Kernel/linker_amd64.ld:3]: we might want to change this value in the future gcc has
```
-mcmodel=kernel

    Generate code for the kernel code model. The kernel runs in the negative 2 GB of the address space. This model has to be used for Linux kernel code.

```
for example, I could not find a clang eqivalent though (they have -mkernel though, but dont say what it does)

File: Bootloader/init.cpp [hunk 2]
```diff
@@ -21,15 +21,16 @@ extern "C" [[noreturn]] void __stack_chk_fail();
...
-extern "C" __attribute__((section(".boot_text"))) void* code64_sel;
+extern "C" u8 gdt64ptr[];
+extern "C" u16 code64_sel;
 extern "C" u64 boot_pml4t[512];
 extern "C" u64 boot_pdpt[512];
 extern "C" u64 boot_pd0[512];
+extern "C" u64 boot_pd0_pts[512 * (MAX_KERNEL_SIZE >> 21)];
 extern "C" u64 boot_pd_kernel[512];
-extern "C" u64 boot_pd_kernel_pts[512 * (KERNEL_PD_OFFSET >> 21)];
+extern "C" u64 boot_pd_kernel_pts[512 * (MAX_KERNEL_SIZE >> 21)];
```
Comment (Hendiadyoin1) [inline Bootloader/init.cpp]: Maybe &1FF here?
Same in the boot.S file

Comment (gunnarbeutner) [reply Bootloader/init.cpp]: Fixed. :)

**Comment by gunnarbeutner** [comment]
What's not fixed (from the other PR): Formatting addresses is still kinda annoying (`#ifdef`s).

**Pull request merged**